L1Trigger/DTTriggerPhase2/src/LateralityCoarsedProvider.cc

```diff
@@ -0,0 +1,1373 @@
+  lat_combinations.push_back({3, {1, 1, 0, -1}, {1, 3, 0, -1}, {{1, 0, 0, -1}, {0, 0, 0, 0}, {0, 0, 0, 0}}});
+  lat_combinations.push_back({3, {1, 1, 0, -1}, {0, 3, 2, -1}, {{1, 1, 0, -1}, {0, 0, 0, 0}, {0, 0, 0, 0}}});
+  lat_combinations.push_back({3, {1, 1, 0, -1}, {2, 3, 0, -1}, {{1, 0, 0, -1}, {0, 0, 0, 0}, {0, 0, 0, 0}}});
+  lat_combinations.push_back({3, {1, 1, 0, -1}, {0, 3, 3, -1}, {{1, 1, 0, -1}, {0, 0, 0, 0}, {0, 0, 0, 0}}});
+  lat_combinations.push_back({3, {1, 1, 0, -1}, {3, 3, 0, -1}, {{1, 0, 0, -1}, {0, 0, 0, 0}, {0, 0, 0, 0}}});
+  lat_combinations.push_back({3, {1, 1, 0, -1}, {3, 2, 0, -1}, {{1, 0, 0, -1}, {0, 0, 0, 0}, {0, 0, 0, 0}}});
+  lat_combinations.push_back({3, {1, 1, 0, -1}, {0, 2, 0, -1}, {{1, 1, 0, -1}, {1, 0, 0, -1}, {0, 0, 0, 0}}});
+  lat_combinations.push_back({3, {1, 1, 0, -1}, {0, 2, 3, -1}, {{1, 1, 0, -1}, {0, 0, 0, 0}, {0, 0, 0, 0}}});
+  lat_combinations.push_back({3, {1, 1, 0, -1}, {1, 2, 0, -1}, {{1, 0, 0, -1}, {0, 0, 0, 0}, {0, 0, 0, 0}}});
+  lat_combinations.push_back({3, {1, 1, 0, -1}, {0, 2, 1, -1}, {{1, 1, 0, -1}, {0, 0, 0, 0}, {0, 0, 0, 0}}});
+  lat_combinations.push_back({3, {1, 1, 0, -1}, {2, 2, 0, -1}, {{1, 0, 0, -1}, {0, 0, 0, 0}, {0, 0, 0, 0}}});
+  lat_combinations.push_back({3, {1, 1, 0, -1}, {0, 2, 2, -1}, {{1, 1, 0, -1}, {0, 0, 0, 0}, {0, 0, 0, 0}}});
+  lat_combinations.push_back({3, {1, 1, 0, -1}, {1, 1, 0, -1}, {{1, 0, 0, -1}, {0, 0, 0, 0}, {0, 0, 0, 0}}});
+  lat_combinations.push_back({3, {1, 1, 0, -1}, {3, 1, 0, -1}, {{1, 0, 0, -1}, {0, 0, 0, 0}, {0, 0, 0, 0}}});
+  lat_combinations.push_back({3, {1, 1, 0, -1}, {0, 1, 0, -1}, {{1, 0, 0, -1}, {1, 1, 0, -1}, {0, 0, 0, 0}}});
+  lat_combinations.push_back({3, {1, 1, 0, -1}, {0, 1, 3, -1}, {{1, 1, 0, -1}, {0, 0, 0, 0}, {0, 0, 0, 0}}});
+  lat_combinations.push_back({3, {1, 1, 0, -1}, {0, 1, 2, -1}, {{1, 1, 0, -1}, {0, 0, 0, 0}, {0, 0, 0, 0}}});
+  lat_combinations.push_back({3, {1, 1, 0, -1}, {2, 1, 0, -1}, {{1, 0, 0, -1}, {0, 0, 0, 0}, {0, 0, 0, 0}}});
+  lat_combinations.push_back({3, {1, 1, 0, -1}, {0, 1, 1, -1}, {{1, 1, 0, -1}, {0, 0, 0, 0}, {0, 0, 0, 0}}});
+  lat_combinations.push_back({3, {1, 1, 0, -1}, {0, 0, 0, -1}, {{1, 1, 0, -1}, {1, 0, 0, -1}, {0, 0, 0, 0}}});
+  lat_combinations.push_back({3, {1, 1, 0, -1}, {3, 0, 0, -1}, {{1, 0, 0, -1}, {0, 0, 0, 0}, {0, 0, 0, 0}}});
+  lat_combinations.push_back({3, {1, 1, 0, -1}, {0, 0, 1, -1}, {{1, 1, 0, -1}, {0, 0, 0, 0}, {0, 0, 0, 0}}});
+  lat_combinations.push_back({3, {1, 1, 0, -1}, {1, 0, 0, -1}, {{1, 0, 0, -1}, {0, 0, 0, 0}, {0, 0, 0, 0}}});
+  lat_combinations.push_back({3, {1, 1, 0, -1}, {0, 0, 2, -1}, {{1, 1, 0, -1}, {0, 0, 0, 0}, {0, 0, 0, 0}}});
+  lat_combinations.push_back({3, {1, 1, 0, -1}, {0, 0, 3, -1}, {{1, 1, 0, -1}, {0, 0, 0, 0}, {0, 0, 0, 0}}});
+  lat_combinations.push_back({3, {1, 1, 0, -1}, {2, 0, 0, -1}, {{1, 0, 0, -1}, {0, 0, 0, 0}, {0, 0, 0, 0}}});
+};
```

Comment:
Is this a direct firmware port? Could a loop or some logic or something be written to simplify this?

Reply:
I am not sure if there is a logic, I must check with the firmware developers....

Reply:
Done